kubeflow/pipeline/pipeline-ui.libsonnet

```diff
@@ -144,6 +144,23 @@
             "list",
           ],
         },
+        {
+          apiGroups: [
+            "kubeflow.org",
+          ],
+          resources: [
+            "viewers",
+          ],
+          verbs: [
+            "create",
+            "get",
+            "list",
+            "watch",
```

Comment:
does it need watch, patch, update?

Reply:
We can be pedantic and remove watch/patch, but I imagine update is useful. I don't think the difference is material once you give permissions to create and delete.

Reply:
It is generally good practice to be pedantic with permission to start with minimum. this way we know exactly what UI does by going through this list.

2 month later i need to scratch my head to figure out why ui need to patch/update an immutable crd instance. wdyt?

Reply:
Missed the comments here. I'll have a separate PR to clean up the patch and watch permissions, both in this file and ml-pipeline-ui-role